### PR TITLE
Abstract Directory Items

### PR DESCRIPTION
**Pull request by Windchild292**
This is the basis of a plan I have for implementing a full abstract directory, and at the very least improves the readability and abstracts the code used in our directories.

Please note this code has undergone a few months worth of testing as the basis for my PRs.

### CODE REVIEW

**Comment by Windchild292** [comment]
Honestly, this is a good point.

I'm not sure on the history, but the goal of this was to standardize and make it easier to parse in the future while not doing any changes outside of these files. Adding new methods to return collections would be fine, but I'd like to do some testing before I submit to see if there's a reason why this was done for now.

**Comment by Windchild292** [comment]
Either way, I'm going to open up this as I've gotten confirmation from a second tester that this is working pretty well. If we want to add methods returning Collections just message me on slack and I'll add it.

File: megamek/src/megamek/common/util/fileUtils/AbstractDirectory.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ */
+public abstract class AbstractDirectory implements Categorized {
+    //region Variable Declarations
+    private String rootName; // The root category name
+
+    /**
+     * A map of the category names to the sub-categories. Please note that this
+     * map includes the root category, if the root category contains any items.
+     */
+    protected TreeMap<String, Categorized> categories = new TreeMap<>(StringUtil.stringComparator());
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply megamek/src/megamek/common/util/fileUtils/AbstractDirectory.java:41]: I'm not certain, but they are how we handle it currently (no change, just abstracted out)

[the remark this answers is not in the document]
Comment (sixlettervariables) [reply megamek/src/megamek/common/util/fileUtils/AbstractDirectory.java:41]: I see that now, okay!

**Review by sixlettervariables — APPROVED** [review]

**Pull request merged**